File: app/src/main/java/com/ataulm/whatsnext/api/Letterboxd.java
package com.ataulm.whatsnext.api;

import com.ataulm.whatsnext.Film;
import com.ataulm.whatsnext.FilmSummary;
import com.ataulm.whatsnext.Token;

import java.io.IOException;
import java.util.List;

@Deprecated // use LetterboxdApi
public interface Letterboxd {

    Token fetchAccessToken(String username, String password) throws IOException;

    Token refreshAccessToken(String refreshToken) throws IOException;

    List<FilmSummary> search(String searchTerm) throws IOException;

    Film film(String letterboxdId, String accessToken) throws IOException;
}

File: letterboxd/src/main/java/com/ataulm/whatsnext/letterboxd/Letterboxd.kt
package com.ataulm.whatsnext.letterboxd

import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface Letterboxd {

    @FormUrlEncoded
    @POST("/auth/token")
    fun accessToken(@Field("username") username: String,
                    @Field("password") password: String,
                    @Field("grant_type") grantType: String = "password"): Observable<Token>

    data class Token(val access_token: String, val refresh_token: String, val expires_in: Long)
}
